src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -392,61 +379,94 @@ private void importFromGcs(String projectId, TableReference tableRef, @Nullable
         LOG.info("Using schema '{}' for the load job config.", loadConfig.getSchema());
       }
 
-      JobConfiguration config = new JobConfiguration();
-      config.setLoad(loadConfig);
-
       // Get the dataset to determine the location
       Dataset dataset =
         bigQueryHelper.getRawBigquery().datasets().get(tableRef.getProjectId(), tableRef.getDatasetId()).execute();
 
+      //Depending on Operation type and no of gcs paths present , trigger suitable BQ job.
+      temporaryTableReference = null;
+      if (operation.equals(Operation.INSERT) &&  gcsPaths.size() <= BQ_IMPORT_MAX_BATCH_SIZE) {
+        // Directly load data into destination table when total no of input paths is loadable into BQ
+        loadConfig.setSourceUris(gcsPaths);
+        loadConfig.setWriteDisposition(writeDisposition);
+        loadConfig.setDestinationTable(tableRef);
+
+        JobConfiguration config = new JobConfiguration();
+        config.setLoad(loadConfig);
+        triggerBigqueryJob(projectId, jobId , dataset, config);
+      } else {
+        // First load the data in a temp table.
+        loadInBatchesInTempTable(tableRef, loadConfig, gcsPaths, projectId, jobId, dataset);
+
+        if (operation.equals(Operation.INSERT)) { // For the case when gcs paths is more than 10000
+          handleInsertOperation(tableRef, writeDisposition, loadConfig.getDestinationEncryptionConfiguration(),
+                                projectId, jobId, dataset, tableExists);
+        } else {
+          handleUpdateUpsertOperation(tableRef, kmsKeyName, getJobIdForUpdateUpsert(conf), conf);
+        }
+      }
+
+      setTemporaryTableExpiration();
```

Comment:
I'd like for the temporary table to get created and the expiration set before the load in batches happens.

Think about it. If there are 2 batches, first succeeds and second fails, the table will not have an expiration set, thus is will stay as a dangling table.

Since I do not want to block the RC5 release, let's merge this and address this in a later PR.

Reply:
This can be achieved by creating the temp table with a TTL and then load records while allowing for the table schema to be relaxed